api/server/httputils/errors.go

```diff
@@ -129,3 +133,24 @@ func statusCodeFromGRPCError(err error) int {
 		return http.StatusInternalServerError
 	}
 }
+
+// statusCodeFromDistributionError returns status code according to registry errcode
+// code is loosely based on errcode.ServeJSON() in docker/distribution
+func statusCodeFromDistributionError(err error) int {
+	switch errs := err.(type) {
+	case errcode.Errors:
+		if len(errs) < 1 {
+			return http.StatusInternalServerError
+		}
+		if err, ok := errs[0].(errcode.ErrorCoder); ok {
+			return err.ErrorCode().Descriptor().HTTPStatusCode
```

Comment:
Nit, this should recurse

Reply:
You mean "loop over `errs`, or call `statusCodeFromDistributionError` here? (as below);

```suggestion
		if _, ok := errs[0].(errcode.ErrorCoder); ok {
			return statusCodeFromDistributionError(errs[0])
		}
```

Reply:
call `statusCodeFromDistributionError`